packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -1536,26 +1536,40 @@ export class MergeTree {
 				const starts = new Set<ObliterateInfo>();
 				const overlapping = new Set<ObliterateInfo>();
 				const findLeftMovedSegment = (seg: ISegment): boolean => {
-					seg.localRefs?.walkReferences(
-						(ref) => {
-							const obliterate = ref.properties?.obliterate as ObliterateInfo | undefined;
-							if (obliterate?.start === ref) {
-								starts.add(obliterate);
-							}
-						},
-						undefined,
-						false,
-					);
+					if (
+						(seg.movedSeq !== undefined &&
```

Comment:
this won't work once we have expansion, as the segment that has the ref won't be moved, so i'm not sure we should add this.

Reply:
an idea could be to put all the obliterate refers in a stored segment set, and do that scan on that, rather than the tree. something like that should probably be a pr of its own.